### PR TITLE
Add unit tests for dbmail/receivers modules

### LINKED ISSUE #4387: Add comprehensive test coverage for untested modules esp/esp/dbmail/receivers
### Summary

Multiple modules in `esp/esp/dbmail/receivers/` have **zero test coverage**. This issue tracks adding unit tests for all previously untested files in this directory and enforcing testing as a PR requirement going forward.

### Problem

After auditing the codebase, the following files in `esp/esp/dbmail/receivers/` were found to have **no tests at all**:

- `classlist.py` 
- `plainlist.py` 
- `sectionlist.py` 
- `useremail.py` 

---

### Proposed Solution

# Changes Required

### New Test Files (4 files)

All test files should be placed in `esp/esp/tests/` with filenames indicating the source file under test:

| Test File | Source Module |
|-----------|--------------|
| `test_dbmail_classlist.py` | `dbmail/receivers/classlist.py` |
| `test_dbmail_plainlist.py` | `dbmail/receivers/plainlist.py` |
| `test_dbmail_sectionlist.py` | `dbmail/receivers/sectionlist.py` |
| `test_dbmail_useremail.py` | `dbmail/receivers/useremail.py` |

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by v0idheaven**
## Description

Adds comprehensive unit tests for email routing receivers in the `esp.dbmail.receivers` module. These tests cover the core email routing logic for class-based, section-based, plain redirect, and user-specific email routing.

**Tests Added:**

- **ClassList receiver** (`test_dbmail_classlist.py`)
  - Routing logic for USE_MAILMAN flag
  - process_nomailman for different user types (teachers, students, class)
  - Edge cases (invalid IDs, case sensitivity, whitespace)
  - process_mailman with proper mocking of mailman functions

- **PlainList receiver** (`test_dbmail_plainlist.py`)
  - Redirect matching and case-insensitive lookup
  - Single and multiple redirect handling
  - Edge cases (no match, empty user)

- **SectionList receiver** (`test_dbmail_sectionlist.py`)
  - Section-specific email routing
  - User type filtering for sections
  - Mailman integration for section lists

- **UserEmail receiver** (`test_dbmail_useremail.py`)
  - Teacher vs non-teacher routing
  - List-Id header handling
  - Direct send functionality

These tests improve code coverage and ensure email routing logic works correctly across different configurations and edge cases.

## Related Issue

Closes #4387

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactor / cleanup
- [ ] CI/CD or build change

## Testing

- [x] Existing tests pass
- [x] New tests added (if applicable)
- [ ] Manually verified

## AI Disclosure

N/A

## Checklist

- [x] Self-reviewed the code
- [ ] Updated documentation (if needed)
- [x] No new warnings or errors introduced

### CODE REVIEW

File: esp/esp/dbmail/tests/test_dbmail_classlist.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+            handler.process(None, self.cls.id, 'class')
+            mock_m.assert_called_once_with(None, self.cls.id, 'class')
+            mock_nm.assert_not_called()
+
+    @override_settings(USE_MAILMAN=False)
+    def test_invalid_class_id(self):
+        handler = _make_handler()
+        handler.process_nomailman(None, -9999, 'class')
+        self.assertFalse(handler.send,
+                         "send should remain False for an invalid class ID")
```
[the remark this answers is not in the document]
Comment (v0idheaven) [reply esp/esp/dbmail/tests/test_dbmail_classlist.py]: changed exception handling to ObjectDoesNotExist, which catches both ClassSubject.DoesNotExist and ESPUser.DoesNotExist

File: esp/esp/dbmail/tests/test_dbmail_classlist.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        handler.process_mailman(None, self.cls.id, 'class')
+        self.assertFalse(handler.send,
+                         "send should remain False when mailman_moderator is not configured")
+
+    @override_settings(USE_MAILMAN=True, DEFAULT_EMAIL_ADDRESSES={'mailman_moderator': 'mod@example.com'})
+    def test_mailman_invalid_class_id(self):
+        handler = _make_handler()
+        handler.process_mailman(None, -9999, 'class')
+        self.assertFalse(handler.send,
+                         "send should remain False for invalid class ID in mailman mode")
```
[the remark this answers is not in the document]
Comment (v0idheaven) [reply esp/esp/dbmail/tests/test_dbmail_classlist.py]: Changed exception handling to ObjectDoesNotExist, which catches both ClassSubject.DoesNotExist and ESPUser.DoesNotExist

File: esp/esp/dbmail/tests/test_dbmail_classlist.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    @patch('esp.dbmail.receivers.classlist.add_list_member')
+    @patch('esp.dbmail.receivers.classlist.add_list_members')
+    @patch('esp.dbmail.receivers.classlist.apply_list_settings')
+    @patch('esp.dbmail.receivers.classlist.Site')
+    def test_mailman_no_archive_address(self, mock_site, mock_apply, mock_add_members,
+                                        mock_add_member, mock_load, mock_create):
+        mock_site.objects.get_current.return_value.domain = 'example.com'
+
+        handler = _make_handler()
+        handler.process_mailman(None, self.cls.id, 'class')
```
[the remark this answers is not in the document]
Comment (v0idheaven) [reply esp/esp/dbmail/tests/test_dbmail_classlist.py]: Added mocks for set_list_moderator_password and send_mail

File: esp/esp/dbmail/tests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from __future__ import absolute_import
+
+from esp.dbmail.tests.test_dbmail_plainlist import PlainListTest
+from esp.dbmail.tests.test_dbmail_useremail import UserEmailTest
+from esp.dbmail.tests.test_dbmail_classlist import ClassListTest
+from esp.dbmail.tests.test_dbmail_sectionlist import SectionListTest
```
[the remark this answers is not in the document]
Comment (v0idheaven) [reply esp/esp/dbmail/tests/__init__.py]: Removed all imports from __init__.py to avoid interfering with test discovery

**Comment by v0idheaven** [comment]
Hi @willgearty , @kkbrum , @hwatheod , this PR is ready for review whenever you get a chance!
